### PR TITLE
Bug 1478420 - Fix reference publishing

### PR DESCRIPTION
**Pull request by djmitche**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -27,7 +27,8 @@ services, with the following structure:
 
 This is a subset of the
 [taskcluster-lib-docs](https://github.com/taskcluster/taskcluster-lib-docs)
-documentation tarball format, and `metadata.json` is defined there.
+documentation tarball format, and `metadata.json` is defined there.  The library
+will load `exchanges.json` as an alternatie to `events.json`, if present.
```
Comment (helfi92) [inline README.md:31]: `s/alternatie/alternative/`.

File: README.md [hunk 2]
```diff
@@ -15,7 +15,7 @@ services, with the following structure:
 ├── myservice
 │   ├── metadata.json
 │   ├── references
-│   │   ├── exchanges.json
+│   │   ├── events.json
```
Comment (helfi92) [inline README.md:18]: Why this change?

**Review by helfi92 — APPROVED** [review]
Is there a document I could read to get more information about how this service works in general?

**Comment by djmitche** [comment]
It doesn't do much - the README summarizes it.  It's basically static content containing a bunch of .json files.

File: README.md (hunk 2 above)
Comment (djmitche) [reply README.md:18]: The tarball format contains events.json, not exchanges.json.  I confused myself when I first wrote this because the references site uses exchanges.json.  The two things used to be separate, so the naming discrepancy didn't matter.  At this point, it's easier for tc-references to change the name (take events.json as input, produce exchagnes.json as output) than to change either one.

**Comment by djmitche** [comment]
Probably the more interesting bit to look at is how this service is used in the cluster build process (in https://github.com/taskcluster/taskcluster-builder).  Basically, as each service and library is built, its docs tarballs are untarred into a global "docsDir".  Then tc-references is bundled up with those docs into a Docker image.  Its `yarn build` runs on container startup, before nginx is started, and creates the appropriate files to serve schemas and references -- with the correct rootUrl.  The results are at https://taskcluster.imbstack.com/references/ and https://taskcluster.imbstack.com/schemas/

Docs is similar, but its build process occurs at docker image creation time, because it does not need a rootUrl to build.
